pest/src/position.rs

```diff
@@ -321,12 +332,8 @@ impl<'i> Position<'i> {
     #[inline]
     pub(crate) fn match_insensitive(&mut self, string: &str) -> bool {
         let matched = {
-            // Matching is safe since, even if the string does not fall on UTF-8 borders, that
-            // particular slice is only used for comparison which will be handled correctly.
-            let slice = unsafe { str::from_utf8_unchecked(&self.input[self.pos..]) };
-
-            if slice.is_char_boundary(string.len()) {
-                let slice = unsafe { slice.get_unchecked(0..string.len()) };
+            let slice = &self.input[self.pos..];
```

Comment:
This whole function should (as noted above) work in bytes. The whole thing should probably have a go-over to see where our "cursor" is currently pointing inside of a Unicode codepoint to determine if there's a different way to handle it. (And benchmark along the way.)

```rust
fn match_insensitive(&mut self, string: &str) -> bool {
    let matched = {
        let slice = &self.input.as_bytes()[self.pos..];
        if let Some(slice) = slice.get(0..string.len()) {
            slice.eq_ignore_ascii_case(&string.as_bytes())
        } else {
            false
        }
    };
    if matched {
        self.pos += string.len();
    }
    matched

Reply:
Is this something we want to do before merging this?

Reply:
No, this implementation is good as is for now.

Future profiling could definitely take a pass through here for what functions are slicing strings and could be slicing bytes instead, though.